File: rag/src/api/schemas/llm.py
import uuid

from pydantic import BaseModel


class LLMCreate(BaseModel):
    model_name: str
    api_key: str  # plain string - encryption happens in the service layer
    is_active: bool = True


class LLMUpdate(BaseModel):
    model_name: str | None = None
    api_key: str | None = None
    is_active: bool | None = None


class LLMRead(BaseModel):
    id: uuid.UUID
    model_name: str
    is_active: bool

    model_config = {"from_attributes": True}

File: web/src/api/types/llm.ts
export type LLMRead = {
  id: string;
  model_name: string;
  is_active: boolean;
  user_id: string;
};

export type LLMCreate = {
  model_name: string;
  api_key: string;
  is_active?: boolean;
};

export type LLMUpdate = {
  model_name?: string;
  api_key?: string;
  is_active?: boolean;
};